lib/ddtrace/profiling/transport/http.rb

```diff
@@ -65,20 +65,21 @@ def self.default_headers
         end
 
         private_class_method def self.default_adapter
-          :net_http
+          Datadog::Ext::Transport::HTTP::ADAPTER
         end
 
         private_class_method def self.configure_for_agent(transport, profiling_upload_timeout_seconds:, agent_settings:)
           apis = API.agent_defaults
 
           transport.adapter(
-            default_adapter,
-            agent_settings.hostname,
-            agent_settings.port,
+            agent_settings.adapter,
+            hostname: agent_settings.hostname,
+            port: agent_settings.port,
```

Comment:
> should we just pass in the whole agent_settings object?

The adapters' initializers are indirectly public API: https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#using-the-nethttp-adapter

For example, whatever is passed after `:net_http` in `t.adapter :net_http, '127.0.0.1', 8126, { timeout: 1 }` is passed to the initializer as is.
I don't see value in wrapping it in an AgentSettings object in such scenario, unless you think otherwise.

> for instance the net_http ones where we needed to add uds_path and filepath references because the class needs to receive, and then ignore then.

I can't find a case like this in the PR, what part exactly are you referring to?

Reply:
> The adapters' initializers are indirectly public API

Yup, I'm aware. See my TL;DR above where I "share a bit more of my thinking". My suggestion here would be kind of the next step in the agent settings, which is to make them go all the way to the transport.

I'm aware that we would still need to support for a time whatever arguments are allowed to be called from the transport proc.

> I can't find a case like this in the PR, what part exactly are you referring to?

I mean that in the `profiling/transport/http_spec.rb` and `transport/http_spec.rb` files, we needed to add `uds_path` to specs (or parts of specs) that otherwise are http-specific.

Reply:
Done! Take a look at see if you like the changes. `AgentSettings` now goes all the way to the adapters.

Reply:
(posting a comment here so github stops changing my focus to this comment draft for no reason)